metadata_please/pep621.py

```diff
@@ -0,0 +1,39 @@
+try:
+    import tomllib as toml
+except ImportError:
+    import toml  # type: ignore[no-redef,unused-ignore]
+from pathlib import Path
+
+from packaging.utils import canonicalize_name
+
+from .types import BasicMetadata
+
+
+def merge_markers(extra_name: str, value: str) -> str:
+    if ";" not in value:
+        return f'{value} ; extra == "{extra_name}"'
+    else:
+        a, _, b = value.partition(";")
+        a = a.strip()
+        b = b.strip()
+        return f'{a} ; ({b}) and extra == "{extra_name}"'
+
+
+def from_pep621_checkout(path: Path) -> bytes:
+    data = (path / "pyproject.toml").read_text()
+    doc = toml.loads(data)
+
+    buf: list[str] = []
+    for dep in doc.get("project", {}).get("dependencies", ()):
+        buf.append(f"Requires-Dist: {dep}\n")
```

Comment:
Does this need a `merge_markers(dep)`?

Reply:
Nope, in this case `dep` already has embedded markers in its string, and we're not adding it in the context of an extra (that's the following couple lines).